ui/src/app/utils/runtime-presets.ts

```diff
@@ -70,7 +67,19 @@ export const applyPresetOverride = (runtime) => {
     const { gceConfig, gceWithPdConfig, dataprocConfig } =
       runtimePresets[runtimePresetKey].runtimeTemplate;
 
-    return { ...runtime, gceConfig, gceWithPdConfig, dataprocConfig };
+    return {
+      ...runtime,
+      gceConfig,
+      // restore the original PD name, which will cause a creation request to attach it to the new runtime
+      gceWithPdConfig:
+        gceWithPdConfig &&
+        fp.set(
+          ['persistentDisk', 'name'],
+          runtime.gceWithPdConfig?.persistentDisk?.name,
```

Comment:
Just an FYI, I generally try to use as few of these as possible. For example, if it were the case that gceWithPdConfig can be null, but if it isn't then it will definitely have a persistentDisk, you could write `runtime.gceWithPdConfig?.persistentDisk.name` which gives the reader a little more information about the structure of the data.

Reply:
My instinct is to use as *many* as possible :)

But I take your point - that's a good practice when I'm confident about the data.